### DISCUSSION TITLE
Radio automatisch starten nach Stromzufuhr / Boot

### DISCUSSION THREAD

**Discussion by mcbo007** [beabel/radiowecker#55] [Fragen & Antworten / Q&A] [answered]
Hi, was muss ich im Code ändern,
damit das Radio sofort nach dem Boot / Stromzufuhr aktiviert ist,
ohne es erst per Touch einschalten zu müssen?

Gruß

Marcel

**Comment by beabel** [comment, marked as answer]
versuche es mal mit 

```
toggleRadio(radio);
```
im setup loop am besten direkt nach 
```
    showClock();
```
 HINWEIS ungetestet!!!!

**Reply by mcbo007** [reply]
Danke, funktioniert !!! 👍

Gruß

Marcel

**Reply by kurtel57** [reply]
Hallo, 
ich versuche verzweifelt rauszufinden wo das "toggleRadio(radio);" hin muss. Könnte ich die Zeilennummer in der IDE bekommen?
Vielen Dank

**Reply by mcbo007** [reply]
Hi, das ist schon eine Weile her ;-)
Bei mir war es Zeile 181.

`  if (connected) {  //successful connection
    //setup real time clock
    configTzTime("CET-1CEST,M3.5.0/03,M10.5.0/03", ntp.c_str());
    //show date and time and the name of the station
    delay(500);
    //fill timestarukture ti, minutes and weekday with now
    getLocalTime(&ti);
    minutes = ti.tm_hour * 60 + ti.tm_min;
    weekday = ti.tm_wday;
    Serial.println("Start");
    //if alarm is on get date and time for next alarm
    if (pref.isKey("alarmon") && pref.getBool("alarmon")) findNextAlarm();
    //Display time and next alarm if one is set
    showClock();
    toggleRadio(radio); //m.boy startet das Radio nach Stromzufuhr
  } else {  //connection not successful
    //we have no connection. A message will be shown on the display
    displayClear();
    displayMessage(5, 10, 310, 30, "Not connected", ALIGNCENTER, true, ILI9341_RED, ILI9341_BLACK, 1);
    displayMessage(5, 60, 310, 30, "connect to SSID radioweckerconf", ALIGNCENTER, false, ILI9341_WHITE, ILI9341_BLACK, 1);
    displayMessage(5, 90, 310, 30, "Configuration on IP: 192.168.4.1", ALIGNCENTER, false, ILI9341_WHITE, ILI9341_BLACK, 1);
    //remember current tick count to allow
    //a retry after five minutes disconnected
    discon = millis();
  }
`

Gruß

Marcel

**Comment by kurtel57** [comment]
Hallo Marcel,

danke für die schnelle Antwort. Werde es ausprobieren.

Gruß

Bernd

 

Von: mcbo007 ***@***.***> 
Gesendet: Montag, 23. Februar 2026 18:35
An: beabel/radiowecker ***@***.***>
Cc: kurtel57 ***@***.***>; Comment ***@***.***>
Betreff: Re: [beabel/radiowecker] Radio automatisch starten nach Stromzufuhr
/ Boot (Discussion #55)

 

Hi, das ist schon eine Weile her ;-)
Bei mir war es Zeile 181.

if (connected) { //successful connection //setup real time clock
configTzTime("CET-1CEST,M3.5.0/03,M10.5.0/03", ntp.c_str()); //show date and
time and the name of the station delay(500); //fill timestarukture ti,
minutes and weekday with now getLocalTime(&ti); minutes = ti.tm_hour * 60 +
ti.tm_min; weekday = ti.tm_wday; Serial.println("Start"); //if alarm is on
get date and time for next alarm if (pref.isKey("alarmon") &&
pref.getBool("alarmon")) findNextAlarm(); //Display time and next alarm if
one is set showClock(); toggleRadio(radio); //m.boy startet das Radio nach
Stromzufuhr } else { //connection not successful //we have no connection. A
message will be shown on the display displayClear(); displayMessage(5, 10,
310, 30, "Not connected", ALIGNCENTER, true, ILI9341_RED, ILI9341_BLACK, 1);
displayMessage(5, 60, 310, 30, "connect to SSID radioweckerconf",
ALIGNCENTER, false, ILI9341_WHITE, ILI9341_BLACK, 1); displayMessage(5, 90,
310, 30, "Configuration on IP: 192.168.4.1", ALIGNCENTER, false,
ILI9341_WHITE, ILI9341_BLACK, 1); //remember current tick count to allow //a
retry after five minutes disconnected discon = millis(); }

Gruß

Marcel


Reply to this email directly, view it on GitHub
<https://github.com/beabel/radiowecker/discussions/55#discussioncomment-1590
0102> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/BIJ6UQIHRBHUZ47WR3SMIX34N
M22PAVCNFSM6AAAAACV474GEWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTK
OJQGAYTAMQ> .
You are receiving this because you commented.
<https://github.com/notifications/beacon/BIJ6UQOZBHJMLTA3HIQXIAT4NM22PA5CNFS
M6AAAAACV474GEWWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQA6KO4M.gif> Message ID:
***@***.***
***@***.***>
>
